sys/random/mersenne.c

```diff
@@ -129,7 +133,12 @@ uint32_t genrand_uint32(void)
     y ^= (y << 7) & 0x9d2c5680UL;
     y ^= (y << 15) & 0xefc60000UL;
     y ^= y >> 18;
+
+#ifdef MODULE_ENTROPY
+    return y + entropy;
```

Comment:
Hi, I gave it a thought, and indeed, adding a random variable to another, has nefarious consequences (or at best not the expected ones I think).
For instance the sum of two uniformly distributed variables, is not uniform anymore: 
http://math.stackexchange.com/questions/357672/density-of-sum-of-two-uniform-random-variables-0-1
http://www.dartmouth.edu/~chance/teaching_aids/books_articles/probability_book/Chapter7.pdf figure 7.2.

Assuming "y" and "entropy" variables are random. After adding them we have a probability of 1/2 of having a value >= MaxInt. Better not rely on how overflow is handled by each platform.